plugin/vcs/gitlab/gitlab.go

```diff
@@ -479,84 +481,111 @@ func getRoleAndMappedRole(accessLevel int32) (gitLabRole ProjectRole, bytebaseRo
 	return "", ""
 }
 
-// FetchRepositoryActiveMemberList fetch all active members of a repository
+// FetchRepositoryActiveMemberList fetches all active members of a repository.
+//
+// Docs: https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project
 func (p *Provider) FetchRepositoryActiveMemberList(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID string) ([]*vcs.RepositoryMember, error) {
-	// Official API doc: https://docs.gitlab.com/14.6/ee/api/members.html
-	url := fmt.Sprintf("%s/projects/%s/members/all", p.APIURL(instanceURL), repositoryID)
-	code, body, err := oauth.Get(
-		ctx,
-		p.client,
-		url,
-		&oauthCtx.AccessToken,
-		tokenRefresher(
-			instanceURL,
-			oauthContext{
-				ClientID:     oauthCtx.ClientID,
-				ClientSecret: oauthCtx.ClientSecret,
-				RefreshToken: oauthCtx.RefreshToken,
-			},
-			oauthCtx.Refresher,
-		),
-	)
-	if err != nil {
-		return nil, errors.Wrap(err, "GET")
-	}
-
-	if code == http.StatusNotFound {
-		return nil, common.Errorf(common.NotFound, fmt.Errorf("failed to fetch repository members from GitLab instance %s", instanceURL))
-	} else if code >= 300 {
-		return nil, fmt.Errorf("failed to read repository members from GitLab instance %s, status code: %d",
-			instanceURL,
-			code,
+	// Preallocate the slice to avoid multiple allocations when appending the first
+	// 100 results to the slice.
+	allMembers := make([]RepositoryMember, 0, 100)
```

Comment:
var allMembers []RepositoryMember

Reply:
Please see the comment right above this line, is that you don't think it's worth it?

Reply:
Reverted in https://github.com/bytebase/bytebase/pull/1423#:~:text=fix%3A%20revert%20preallocate.